src/blame.rs

```diff
@@ -0,0 +1,289 @@
+    pub fn orig_start_line(&self) -> usize {
+        unsafe { (*self.raw).orig_start_line_number as usize }
+    }
+
+    /// Returns path to the file where this hunk originated.
+    pub fn path(&self) -> &Path {
+        unsafe {
+            util::bytes2path(::opt_bytes(self, (*self.raw).orig_path).unwrap())
+        }
+    }
+
+    /// Tests whether this hunk has been tracked to a boundary commit.
+    pub fn is_boundary(&self) -> bool {
+        unsafe {
+            (*self.raw).boundary == 1
+        }
+    }
+
+    /// Returns number of lines in this hunk.
+    pub fn len(&self) -> usize {
+        unsafe {
+            (*self.raw).lines_in_hunk as usize
+        }
+    }
+}
+
+impl BlameOptions {
+
+    /// Initialize options
+    pub fn new() -> BlameOptions {
+        let mut opts = BlameOptions {
+            raw: unsafe { mem::zeroed() },
+        };
+        assert_eq!(unsafe {
+            raw::git_blame_init_options(&mut opts.raw,
+                                        raw::GIT_BLAME_OPTIONS_VERSION)
+        }, 0);
+        opts
+    }
+
+    fn flag(&mut self, opt: u32, val: bool) -> &mut BlameOptions {
+        if val {
+            self.raw.flags |= opt;
+        } else {
+            self.raw.flags &= !opt;
+        }
+        self
+    }
+
+    /// Same file
+    pub fn track_copies_same_file(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_FILE, opt)
+    }
+
+    /// Same commit moves
+    pub fn track_copies_same_commit_moves(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_COMMIT_MOVES, opt)
+    }
+
+    /// Same commit copies
+    pub fn track_copies_same_commit_copies(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_COMMIT_COPIES, opt)
+    }
+
+    /// Any commit copies
+    pub fn track_copies_any_commit_copies(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_ANY_COMMIT_COPIES, opt)
+    }
+
+    /// First parent
+    pub fn first_parent(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_FIRST_PARENT, opt)
+    }
+
+    /// Raw pointer
+    pub unsafe fn raw(&mut self) -> *mut raw::git_blame_options {
+        &mut self.raw as *mut _
+    }
+
+    /// Setter for the id of the newest commit to consider.
+    pub fn set_newest_commit(&mut self, id: Oid) -> &mut BlameOptions {
+        unsafe { self.raw.newest_commit = *id.raw(); }
+        self
+    }
+
+    /// Setter for the id of the oldest commit to consider.
+    pub fn set_oldest_commit(&mut self, id: Oid) -> &mut BlameOptions {
+        unsafe { self.raw.oldest_commit = *id.raw(); }
+        self
+    }
+
+    /// Get the newest commit OID.
+    pub fn newest_commit(&self) -> Oid {
+        unsafe { Binding::from_raw(&self.raw.newest_commit as *const _) }
+    }
+
+    /// Get the oldest commit OID.
+    pub fn oldest_commit(&self) -> Oid {
```

Comment:
Looking at the [current source](https://github.com/libgit2/libgit2/blob/25f355cbedb2320ca5c4db5e02d1ab9ca7ac9cab/src/blame.c#L354-L382) while `const` is not located, the [only usage](https://github.com/libgit2/libgit2/blob/25f355cbedb2320ca5c4db5e02d1ab9ca7ac9cab/src/blame.c#L365) does indeed [have the `const` qualifier](https://github.com/libgit2/libgit2/blob/25f355cbedb2320ca5c4db5e02d1ab9ca7ac9cab/src/blame.c#L184).

I also think that the example in the repo is probably just using that variable as a proxy for holding other data (e.g. looking at the value set previously) rather than relying on `git_file_blame` to actually modify it.